src/frequenz/client/common/_types.py

```diff
@@ -0,0 +1,1022 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Module to define the types used with the common client"""
+
+from __future__ import annotations  # required for constructor type hinting
+
+import enum
+import logging
+from datetime import datetime
+from dataclasses import dataclass
+from typing import List
+
+from frequenz.api.common.v1.metrics import metric_sample_pb2, bounds_pb2
```

Comment:
Discussed this in today's meeting and decided to not go with this simplified approach, where we have one single module for all types. Instead we want to keep the protobuf structure, i.e. have separate `metrics`, `microgrid` and `pagination` modules.